packages/core/src/binary.rs

```diff
@@ -219,7 +223,11 @@ impl<'de> Deserialize<'de> for Binary {
     where
         D: Deserializer<'de>,
     {
-        deserializer.deserialize_str(Base64Visitor)
+        if deserializer.is_human_readable() {
+            deserializer.deserialize_str(Base64Visitor)
+        } else {
+            panic!("Binary is only intended to be used with JSON serialization for now")
```

Comment:
Could we add that in all places?

```suggestion
            panic!("Binary is only intended to be used with JSON serialization for now. If you are hitting this panic, please open an issue at https://github.com/CosmWasm/cosmwasm describing your use case.")
```